accounting/invoice/choices.py

```diff
@@ -36,4 +38,146 @@ class InvoiceTemplate(DjangoChoices):
+        """
+        return cls.number.format(number=1)
+
+    @classmethod
+    def default_year_month_number(cls):
+        """
+        Return the default value given to the `default` choice.
+        """
+        return '{}-{}'.format(cls.default_default(), cls.default_number())
+
+    @classmethod
+    def default_opencraft_year_month(cls):
+        """
+        Return the default value given to the `default` choice.
+        """
+        return 'OC-{}'.format(cls.default_default())
+
+    @classmethod
+    def default_opencraft_number(cls):
+        """
+        Return the default value given to the `default` choice.
+        """
+        return 'OC-{}'.format(cls.default_number())
+
+    @classmethod
+    def increment_default(cls, value):
+        """
+        Increment the number for the 'default' numbering scheme.
+
+        Assumes `value` is in the format of the 'default' numbering scheme.
+        """
+        # Turn the input value string into a datetime object.
+        datetime_obj = datetime.datetime.strptime(value, cls.default)
+        # Increment it one month forward.
+        incremented_obj = datetime_obj + datetime.timedelta(days=32)
+        # Turn the object back into a string using the numbering scheme.
+        return incremented_obj.strftime(cls.default)
+
+    @classmethod
+    def increment_number(cls, value):
+        """
+        Increment the number for the 'number' numbering scheme.
+
+        Assumes `value` is in the format of the 'number' numbering scheme.
+        """
+        return cls.number.format(number=int(value) + 1)
+
+    @classmethod
+    def increment_year_month_number(cls, value):
+        """
+        Increment the number for the 'year_month_number' numbering scheme.
+
+        Assumes `value` is in the format of the 'year_month_number' numbering scheme.
+        """
+        date, number = value.rsplit('-', 1)
+        incremented_date = cls.increment_default(date)
+        incremented_number = cls.increment_number(number)
+        return '-'.join([incremented_date, incremented_number])
+
+    @classmethod
+    def increment_opencraft_year_month(cls, value):
+        """
+        Increment the number for the 'opencraft_year_month' numbering scheme.
+
+        Assumes `value` is in the format of the 'opencraft_year_month' numbering scheme.
+        """
+        prefix, date = value.split('-', 1)
+        incremented_date = cls.increment_default(date)
+        return '-'.join([prefix, incremented_date])
+
+    @classmethod
+    def increment_opencraft_number(cls, value):
+        """
+        Increment the number for the 'opencraft_number' numbering scheme.
+
+        Assumes `value` is in the format of the 'opencraft_number' numbering scheme.
+        """
+        prefix, number = value.split('-', 1)
+        incremented_number = cls.increment_number(number)
+        return '-'.join([prefix, incremented_number])
+
+    @classmethod
+    def increment_value(cls, choice, value):
```

Comment:
@UmanShahzad Sorry for not catching this earlier, but I think it would be good to rename the second parameter to `numbering_scheme`; it's a bit more specific than `choice`.

Reply:
:+1: